libethereum/SnapshotStorage.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+	This file is part of cpp-ethereum.
+
+	cpp-ethereum is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	cpp-ethereum is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "SnapshotStorage.h"
+#include <libdevcore/CommonIO.h>
+#include <libdevcore/SHA3.h>
+
+#include <boost/filesystem.hpp>
+#include <snappy.h>
+
+using namespace dev;
+using namespace eth;
+
+namespace
+{
+
+std::string snappyUncompress(std::string const& _compressed)
+{
+	size_t uncompressedSize = 0;
+	if (!snappy::GetUncompressedLength(_compressed.data(), _compressed.size(), &uncompressedSize))
+		BOOST_THROW_EXCEPTION(FailedToGetUncompressedLength());
+
+	std::vector<char> uncompressed(uncompressedSize);
+	if (!snappy::RawUncompress(_compressed.data(), _compressed.size(), uncompressed.data()))
```

Comment:
Because we are using Hunter for Snappy package, you don't need to use RawUncompress any more (Hunter guarantees the same ABI because it uses exactly the same compiler to compile dependencies).

If you want to stick with it, you don't have to use vector. You can do the same with string directly. But still you will init the string with zeros, what is not needed.

Reply:
I'll use `Uncompress` with `string`, it'simpler.
(vector was needed here because string can give only const pointer to it's data)